packages/components/src/systems/RecentOrders/index.tsx

```diff
@@ -0,0 +1,189 @@
+  // if provider chooses not to combine the order, call this to create a new order
+  createOrder?: () => void;
+  // if the user clicks "continue" on the duplicate dialog, we need to call the callback to add to draft prescriptions
+  duplicateDialogContinueCb?: () => void;
+  // reference for the order that the user clicked
+  orderWithIssue?: Order;
+  // reference for the fill with a duplicate treatment name
+  duplicateFill?: Fill;
+};
+
+type RecentOrdersActions = {
+  setIsCombineDialogOpen: (
+    isOpen: boolean,
+    createOrder?: () => void,
+    draftPrescriptions?: DraftPrescription[]
+  ) => void;
+  setIsDuplicateDialogOpen: (
+    isOpen: boolean,
+    duplicateFill?: Fill,
+    continueCb?: () => void
+  ) => void;
+  setIsIssueDialogOpen: (isOpen: boolean, orderWithIssue?: Order) => void;
+  checkDuplicateFill: (treatmentName: string) => Fill | undefined;
+  hasRoutingOrder: () => boolean;
+};
+
+type RecentOrdersContextValue = [RecentOrdersState, RecentOrdersActions];
+
+const RecentOrdersContext = createContext<RecentOrdersContextValue>([
+  {
+    orders: [],
+    isCombineDialogOpen: false,
+    isDuplicateDialogOpen: false,
+    isIssueDialogOpen: false
+  },
+  {
+    setIsCombineDialogOpen: () => undefined,
+    setIsDuplicateDialogOpen: () => undefined,
+    setIsIssueDialogOpen: () => undefined,
+    checkDuplicateFill: () => undefined,
+    hasRoutingOrder: () => false
+  }
+]);
+
+interface SDKProviderProps {
+  patientId: string;
+  children: JSXElement;
+}
+
+function RecentOrders(props: SDKProviderProps) {
+  const client = usePhotonClient();
+  const [state, setState] = createStore<RecentOrdersState>({
+    orders: [],
+    isCombineDialogOpen: false,
+    isDuplicateDialogOpen: false,
+    isIssueDialogOpen: false,
+    duplicateDialogContinueCb: () => undefined
+  });
+
+  const value: RecentOrdersContextValue = [
+    state,
+    {
+      setIsCombineDialogOpen(isOpen, createOrder, draftPrescriptions) {
+        setState({
+          isCombineDialogOpen: isOpen,
+          ...(createOrder ? { createOrder } : { createOrderCb: undefined }),
+          ...(draftPrescriptions ? { draftPrescriptions } : { draftPrescriptions: [] })
+        });
+      },
+      setIsDuplicateDialogOpen(isOpen, duplicateFill, continueCb) {
+        setState({
+          isDuplicateDialogOpen: isOpen,
+          ...(continueCb && { duplicateDialogContinueCb: continueCb }),
+          ...(duplicateFill && { duplicateFill })
+        });
+      },
+      setIsIssueDialogOpen(isOpen, orderWithIssue) {
+        setState({
+          isIssueDialogOpen: isOpen,
+          ...(orderWithIssue && { orderWithIssue }),
+          ...(!isOpen && { duplicateFill: undefined, orderWithIssue: undefined })
+        });
+      },
+      checkDuplicateFill(treatmentName) {
+        return state.orders
+          .map((order) => order.fills.find((fill) => fill.treatment.name === treatmentName))
+          .find((fill) => fill !== undefined);
+      },
```

Comment:
Not a big deal, but nit:
```suggestion
      checkDuplicateFill(treatmentName) {
        return state.orders.some(order => order.fills.some(fill => fill.treatment.name === treatmentName))
      },
```

At least I think thats what this is doing?